src/main/java/io/github/opencubicchunks/cubicchunks/cubicgen/customcubic/CustomGeneratorSettingsFixer.java

```diff
@@ -0,0 +1,467 @@
+import net.minecraft.init.Blocks;
+import net.minecraft.world.biome.Biome;
+import net.minecraft.world.gen.ChunkGeneratorSettings;
+
+public class CustomGeneratorSettingsFixer {
+
+    public static final int VERSION = 3;
+
+    private static final String[] standard = { "dirt", "gravel", "granite", "diorite", "andesite", "coalOre", "ironOre",
+            "goldOre", "redstoneOre", "diamondOre", "hillsEmeraldOre", "hillsSilverfishStone", "mesaAddedGoldOre" };
+
+    private static final IBlockState[] standardBlockstates = { Blocks.DIRT.getDefaultState(),
+            Blocks.GRAVEL.getDefaultState(),
+            Blocks.STONE.getDefaultState().withProperty(BlockStone.VARIANT, BlockStone.EnumType.GRANITE),
+            Blocks.STONE.getDefaultState().withProperty(BlockStone.VARIANT, BlockStone.EnumType.DIORITE),
+            Blocks.STONE.getDefaultState().withProperty(BlockStone.VARIANT, BlockStone.EnumType.ANDESITE),
+            Blocks.COAL_ORE.getDefaultState(), Blocks.IRON_ORE.getDefaultState(), Blocks.GOLD_ORE.getDefaultState(),
+            Blocks.REDSTONE_ORE.getDefaultState(), Blocks.DIAMOND_ORE.getDefaultState(),
+            Blocks.EMERALD_ORE.getDefaultState(),
+            Blocks.MONSTER_EGG.getDefaultState().withProperty(BlockSilverfish.VARIANT, BlockSilverfish.EnumType.STONE),
+            Blocks.GOLD_ORE.getDefaultState() };
+    private static final Biome[][] standardBiomes = { null, // dirt
+            null, // gravel
+            null, // granite
+            null, // diorite
+            null, // andesite
+            null, // coal
+            null, // iron
+            null, // gold
+            null, // redstone
+            null, // diamond
+            { Biomes.EXTREME_HILLS, Biomes.EXTREME_HILLS_EDGE, Biomes.EXTREME_HILLS_WITH_TREES,
+                    Biomes.MUTATED_EXTREME_HILLS, Biomes.MUTATED_EXTREME_HILLS_WITH_TREES }, // emerald
+            { Biomes.EXTREME_HILLS, Biomes.EXTREME_HILLS_EDGE, Biomes.EXTREME_HILLS_WITH_TREES,
+                    Biomes.MUTATED_EXTREME_HILLS, Biomes.MUTATED_EXTREME_HILLS_WITH_TREES }, // monster
+                                                                                             // egg
+            { Biomes.MESA, Biomes.MESA_CLEAR_ROCK, Biomes.MESA_ROCK, Biomes.MUTATED_MESA,
+                    Biomes.MUTATED_MESA_CLEAR_ROCK, Biomes.MUTATED_MESA_ROCK },// mesa
+                                                                               // gold
+    };
+
+    public static String fixGeneratorOptions(String generatorOptionsToFix) {
+        Gson gson = CustomGeneratorSettings.gson(false);
+
+        JsonReader oldReader = new JsonReader(new StringReader(generatorOptionsToFix));
+        JsonObject oldRoot = new JsonParser().parse(oldReader).getAsJsonObject();
+
+        JsonReader newReader = new JsonReader(new StringReader(""));
+        JsonObject newRoot = new JsonParser().parse(newReader).getAsJsonObject();
+
+        newRoot.add("version", new JsonPrimitive(VERSION));
```

Comment:
Each fixer method should handle transition from one version to the next. The idea being that every time you update, you only have to care about correct transition from previous version to the next, and not about any previous version. So this should be a hardcoded 3.